File: ASP.Net/Project/Models/Hostler.cs
using System;
using System.Collections.Generic;

namespace Project.Models
{
    public partial class Hostler
    {
        public Hostler()
        {
            Mealallocations = new HashSet<Mealallocation>();
            Roomallocations = new HashSet<Roomallocation>();
        }

        public int HostlerId { get; set; }
        public string Dateofbirth { get; set; } = null!;
        public string Email { get; set; } = null!;
        public string Firstname { get; set; } = null!;
        public string Lastname { get; set; } = null!;
        public string Phonenumber { get; set; } = null!;
        public int? AddressId { get; set; }
        public int UserId { get; set; }

        public virtual Address? Address { get; set; }
        public virtual User User { get; set; } = null!;
        public virtual ICollection<Mealallocation> Mealallocations { get; set; }
        public virtual ICollection<Roomallocation> Roomallocations { get; set; }
    }
}

File: Backend/SpringBoot/HostelManagement/java/com/project/hostelmanagement/entities/Hostler.java
package com.project.hostelmanagement.entities;

import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@Entity
@Table(name = "hostler")
public class Hostler {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Hostler_ID")
    private Integer hostlerid;
    @Column(nullable = false)
    private String firstname;
    @Column(nullable = false)
    private String lastname;
    @Column(nullable = false)
    private String email;
    @Column(nullable = false)
    private String phonenumber;
    @Column(nullable = false)
    private String dateofbirth;
    

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "Address_ID") 
    private Address address;

    @JsonIgnore
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "User_ID",nullable = false)
    private User user;

   @JsonIgnoreProperties("hostler")
    @OneToMany(mappedBy = "hostler", cascade = CascadeType.ALL) 
    private List<RoomAllocation> roomAllocations = new ArrayList();
    
}